connection_go18.go

```diff
@@ -0,0 +1,24 @@
+// +build go1.8
+
+package mysql
+
+import (
+	"context"
+)
+
+func (mc *mysqlConn) Ping(ctx context.Context) error {
+	err := mc.writeCommandPacket(comPing)
+	if err != nil {
+		return err
+	}
+
+	select {
+	case <-ctx.Done():
+		mc.Close()
+		return ctx.Err()
+	default:
+	}
+
+	_, err = mc.readResultOK()
```

Comment:
No.  `readResultOK()`  **waits** OK packet from MySQL server.  It may take long time.
While waiting OK packet, context can be cancelled.  Then, driver should abandon waiting OK.